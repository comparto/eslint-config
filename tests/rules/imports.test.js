describe('imports.js', () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it('should contain webpack.config.js if webpack is included', () => {
    jest.doMock('../../src/utils', () => ({ doesFileExist: () => true }))

    const imports = require('../../src/rules/imports')

    expect(imports.settings['import/resolver'].webpack.config).toBe(
      'webpack.config.js'
    )
  })

  it('should create default configuration', () => {
    jest.doMock('../../src/utils', () => ({ doesFileExist: () => false }))

    const imports = require('../../src/rules/imports')

    expect(imports).toMatchSnapshot()
  })
})
