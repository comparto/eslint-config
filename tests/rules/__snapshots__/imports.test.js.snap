// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`imports.js should create default configuration 1`] = `
Object {
  "plugins": Array [
    "import",
  ],
  "rules": Object {
    "import/default": "error",
    "import/dynamic-import-chunkname": "off",
    "import/export": "error",
    "import/exports-last": "off",
    "import/extensions": Array [
      "error",
      "ignorePackages",
      Object {
        "js": "never",
        "jsx": "never",
        "mjs": "never",
        "ts": "never",
        "tsx": "never",
      },
    ],
    "import/first": "error",
    "import/group-exports": "off",
    "import/max-dependencies": "off",
    "import/named": "error",
    "import/namespace": "error",
    "import/newline-after-import": "error",
    "import/no-absolute-path": Array [
      "error",
      Object {
        "amd": false,
        "commonjs": true,
        "esmodule": true,
      },
    ],
    "import/no-amd": "error",
    "import/no-anonymous-default-export": "off",
    "import/no-commonjs": "off",
    "import/no-cycle": "warn",
    "import/no-default-export": "off",
    "import/no-deprecated": "warn",
    "import/no-duplicates": "error",
    "import/no-dynamic-require": "off",
    "import/no-extraneous-dependencies": "error",
    "import/no-internal-modules": "off",
    "import/no-mutable-exports": "error",
    "import/no-named-as-default": "warn",
    "import/no-named-as-default-member": "warn",
    "import/no-named-default": "error",
    "import/no-named-export": "off",
    "import/no-namespace": "off",
    "import/no-nodejs-modules": "off",
    "import/no-relative-parent-imports": "off",
    "import/no-restricted-paths": "off",
    "import/no-self-import": "error",
    "import/no-unassigned-import": "off",
    "import/no-unresolved": Array [
      "error",
      Object {
        "caseSensitive": true,
        "commonjs": true,
      },
    ],
    "import/no-unused-modules": "off",
    "import/no-useless-path-segments": "off",
    "import/no-webpack-loader-syntax": "error",
    "import/order": Array [
      "error",
      Object {
        "groups": Array [
          Array [
            "builtin",
            "external",
            "internal",
          ],
        ],
        "newlines-between": "always",
      },
    ],
    "import/prefer-default-export": "off",
    "import/unambiguous": "off",
  },
  "settings": Object {
    "import/extensions": Array [
      ".js",
      ".mjs",
      ".jsx",
      ".ts",
      ".tsx",
    ],
    "import/ignore": Array [
      "node_modules",
    ],
    "import/resolver": Object {
      "node": Object {},
      "webpack": Object {
        "config": Object {},
      },
    },
  },
}
`;
